api/types/src/error.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+use std::fmt;
+use warp::{http::StatusCode, reject::Reject};
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct Error {
+    pub code: u16,
+    pub message: String,
+}
+
+impl Error {
+    pub fn new(code: StatusCode, message: String) -> Self {
+        Self {
+            code: code.as_u16(),
+            message,
+        }
+    }
+
+    pub fn from_anyhow_error(code: StatusCode, err: anyhow::Error) -> Self {
+        Self::new(code, err.to_string())
+    }
+
+    pub fn bad_request(err: anyhow::Error) -> Self {
+        Self::from_anyhow_error(StatusCode::BAD_REQUEST, err)
+    }
+
+    pub fn account_not_found(address: &str) -> Self {
+        Self::not_found(format!("could not find account by address: {}", address))
```

Comment:
To help scale out error handling, I would simply have a not found message. As is, you'll have to add a *_not_found method per resource which doesn't add much value. ie: `fn transaction_not_found`.

You could then collapse a lot of this into [`thiserror`](https://github.com/dtolnay/thiserror#example) with one/few handlers and DRY up this.

There may be nuances that prevent this from being as clean as I hope, but I think it's worth researching so we don't fall into a maintenance trap of adding a error handler method per resource.

Reply:
As we don't have many resources, I think it's better to provide good error message VS less code.
Let me move this into accounts.rs, I think it's better to live there.